src/main/java/com/exasol/adapter/dynamodb/mapping/SchemaMapper.java

```diff
@@ -0,0 +1,59 @@
+package com.exasol.adapter.dynamodb.mapping;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import com.exasol.adapter.dynamodb.documentnode.DocumentNode;
+import com.exasol.adapter.dynamodb.documentpath.DocumentPathExpression;
+import com.exasol.adapter.dynamodb.documentpath.DocumentPathIteratorFactory;
+import com.exasol.adapter.dynamodb.documentpath.PathIterationStateProvider;
+import com.exasol.adapter.dynamodb.remotetablequery.RemoteTableQuery;
+import com.exasol.sql.expression.ValueExpression;
+
+/**
+ * This call maps document data to Exasol rows according to {@link SchemaMappingQuery}.
+ */
+@java.lang.SuppressWarnings("squid:S119") // DocumentVisitorType does not fit naming conventions.
+public class SchemaMapper<DocumentVisitorType> {
+    private final SchemaMappingQuery query;
+    private final ValueMapperFactory<DocumentVisitorType> valueMapperFactory;
+
+    /**
+     * Creates a new {@link SchemaMapper} for a query described in queryResultTableSchema
```

Comment:
```suggestion
     * Creates a new {@link SchemaMapper} for a query described in {@code queryResultTableSchema}.
```

Reply:
But I could not find `queryResultTableSchema`, did you mean `query` here?

Reply:
yes. I fixed the comment.